image-ext2.c

```diff
@@ -211,6 +213,7 @@ static cfg_opt_t ext_opts[] = {
 	CFG_STR("usage-type", NULL, CFGF_NONE),
 	CFG_STR("mke2fs-conf", NULL, CFGF_NONE),
 	CFG_STR("mke2fs_conf", NULL, CFGF_NONE),
+	CFG_INT("block-size", 1024, CFGF_NONE),
```

Comment:
Is 1024 a good default? This is used in mke2fs.conf only for 'small' filesystems and is incompatible with dm-verity or discard on many eMMCs.

Reply:
Well, previously this value was hard-coded. So it was a good "default" for years.

I like to keep the previous behavior. Without this option you will get a block-size of 1024 as before.

Reply:
I have read the mke2fs manual again and I see that I got it wrong on first read.

I will change it.

dm-verity is reason for this. My system seems to be small.
:thinking: So, a user provided mke2fs.conf is the preferred solution?

Reply:
Or just use `extraargs`. That's exactly for use-cases like this.

Reply:
This was my first attempt. It does not work well, because `-b` changes the interpretation of size parameter, so with `extraargs = "-b 4096"` the image was 4 times the size.
Because I got an image with 1kB block-size and I saw the 1k hard-code, I thought this is hard-coded block-size, but I got it wrong.

from man-page:
> The  file  system size is specified by fs-size.  If fs-size does not have a suffix, it is interpreted as power-of-two kilobytes, unless the -b blocksize option is specified, in which case fs-size is interpreted as the number of blocksize blocks.   If the fs-size is suffixed by 'k', 'm', 'g', 't' (either upper-case or lower-case), then it is  interpreted  in  power-of-two  kilobytes,  megabytes, gigabytes, terabytes, etc.  If fs-size is omitted, mke2fs will create the file system based on the device size.

Reply:
` If fs-size does not have a suffix, ...` We can change that, right? Just add a `k` to the size and it's always kilobytes.

Reply:
great. This would also give a hint, why the division is made.

BTW: Is alignment to 1k assumed? Would it be worth to ceil here (`(image->size + 1023) / 1024`)?

Reply:
It's the filesystem size so it must actually be blocksize aligned. And mke2fs will truncate in this case, so we should too.